memory/core.py

```diff
@@ -63,91 +68,71 @@ def update(self):
                 self._assemblies_trg_sig()
                 self._type_info_definition_table_trg_sig()
                 self.get_image()
+                self._set_ready_for_updates()
 
-        except Exception:
+        except Exception as _e:
+            # print(f"Memory Core Reloading {type(_e)}")
             self.__init__()
 
     def get_singleton_by_class_name(self, class_name):
         local_class = self.get_class(class_name)
+        if local_class is None or local_class == 0x0:
+            return None
         parent = self.get_parent(local_class)
+        if parent is None or parent == 0x0:
+            return None
         instance_ptr = self.get_field(parent, "instance")
+        if instance_ptr is None or instance_ptr == 0x0:
+            return None
         static_table = self.get_static_table(parent)
+        if static_table is None or static_table == 0x0:
+            return None
         # The bitwise AND is probably not necessary here
         # but i'll try removing it when more stuff is added
         return (static_table + instance_ptr) & 0xFFFFFFFFFFFFFFFF
 
     def get_pointer(self, root, offsets):
```

Comment:
Maybe add a comment here to indicate that `offsets` will be manipulated and changed by this function (pop).